src/betterproto/templates/template.py.j2

```diff
@@ -5,7 +5,17 @@
 {% for i in output_file.python_module_imports|sort %}
 import {{ i }}
 {% endfor %}
+
+{% if output_file.pydantic_dataclasses %}
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from dataclasses import dataclass
+else:
+    from pydantic.dataclasses import dataclass
+{%- else -%}
 from dataclasses import dataclass
+{% endif %}
+
```

Comment:
Is this necessary? I would have thought all typecheckers knew about `@dataclass_transform`now.

Reply:
Depends on the user's type-checker. I had problems on an old mypy; I'd rather generate over-verbose code than code that cannot be CI'd

Reply:
I don't think there's much merit in not supporting mypy<1.0